setup.py

```diff
@@ -236,6 +236,21 @@ def check_linker_need_libatomic():
     return cpp_test.returncode == 0
 
 
+# When building extensions for macOS on a system running macOS 10.14 or newer,
+# make sure they target macOS 10.14 or newer to use C++17 stdlib properly.
+# This overrides the default behavior of distutils, which targets the macOS
+# version Python was built on. You can further customize the target macOS
+# version by setting the MACOSX_DEPLOYMENT_TARGET environment variable before
+# running setup.py.
+if sys.platform == "darwin":
```

Comment:
Do we also need this [here](https://github.com/grpc/grpc/blob/master/src/python/grpcio_observability/setup.py)?

Reply:
Oh good catch. I added this to `grpcio_observability/setup.py` too

Reply:
We don't need it since observability only support Linux for now, but it's fine to add it.

Reply:
Oh that's why I didn't see errors from grpcio_observability build. Thanks for the info.